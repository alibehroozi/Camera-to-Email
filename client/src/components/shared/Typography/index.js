//@flow
import * as React from "react";
import { TypoWrapper } from "./index.style";

type Props = {
  font: { fontSize: number, fontFamily: string, fontWeight: number },
  color: string,
  children: React.Node
};

const Typography = ({ font, color, children }: Props) => {
  return (
    <TypoWrapper textFont={font} textColor={color}>
      {children}
    </TypoWrapper>
  );
};

export default Typography;
