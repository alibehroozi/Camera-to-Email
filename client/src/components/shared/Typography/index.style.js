//@flow
import styled from "styled-components";

export const TypoWrapper = styled.div`
  color: ${({ textColor }) => textColor};
  font-size: ${({ textFont: { fontSize } }) => fontSize}px;
  font-family: "${({ textFont: { fontFamily } }) => fontFamily}";
  font-weight: ${({ textFont: { fontWeight } }) => fontWeight};
`;
