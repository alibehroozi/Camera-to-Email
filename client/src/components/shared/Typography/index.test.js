//@flow
import React from "react";
import Typography from ".";
import renderer from "react-test-renderer";

describe("<Typography />", () => {
  it("checks the text to be existed", () => {
    const wrapper = renderer
      .create(
        <Typography
          color="red"
          font={{
            fontSize: 20,
            fontWeight: 500,
            fontFamily: "Gudea"
          }}
        >
          mieterengel
        </Typography>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
