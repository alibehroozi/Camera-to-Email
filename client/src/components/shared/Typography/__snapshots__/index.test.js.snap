// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> checks the text to be existed 1`] = `
.c0 {
  color: red;
  font-size: 20px;
  font-family: "Gudea";
  font-weight: 500;
}

<div
  className="c0"
>
  mieterengel
</div>
`;
